### PR TITLE
MINOR: Account for different versions in upgrade

### PR DESCRIPTION
**Pull request by bbejeck**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by bbejeck** [comment]
Kicked off branch builder for `StreamsUpgradeTest#metadata_upgrade.test` https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2644/

**Comment by bbejeck** [comment]
ping @guozhangwang @mjsax @vvcephei @ableegoldman @cadonna @abbccdda 

\cc @ijuma

**Comment by bbejeck** [comment]
branch builder for [metadata-upgrade-test](http://confluent-kafka-branch-builder-system-test-results.s3-us-west-2.amazonaws.com/2019-05-28--001.1559088781--bbejeck--MINOR_account_for_different_versions_in_upgrade--e66a86d/report.html) passed

branch builder for [entire-upgrade-system-test](http://confluent-kafka-branch-builder-system-test-results.s3-us-west-2.amazonaws.com/2019-05-29--001.1559103437--bbejeck--MINOR_account_for_different_versions_in_upgrade--e66a86d/report.html) passed

**Comment by bbejeck** [comment]
Java 11 failed `ERROR: Error cloning remote repo 'origin'`
Java 8 failure unrelated

retest this please

File: tests/kafkatest/tests/streams/streams_upgrade_test.py [hunk 1]
```diff
@@ -536,7 +538,8 @@ def do_rolling_bounce(self, processor, counter, current_generation):
                     self.old_processors.remove(processor)
                     self.upgraded_processors.append(processor)
 
-                    log_monitor.wait_until("Kafka version: " + str(DEV_VERSION),
+                    # checking for the dev version which should be the only SNAPSHOT
+                    log_monitor.wait_until("Kafka version.*-SNAPSHOT",
                                            timeout_sec=60,
                                            err_msg="Could not detect Kafka Streams version " + str(DEV_VERSION) + " in " + str(node.account))
```
Comment (cadonna) [inline tests/kafkatest/tests/streams/streams_upgrade_test.py:545]: Shouldn't `str(DEV_VERSION)` in the error message also be changed?

Comment (bbejeck) [reply tests/kafkatest/tests/streams/streams_upgrade_test.py:545]: I wanted to leave the `str(DEV_VERSION)` as is so it's more clear in the logs if an error occurs

**Comment by bbejeck** [comment]
updated to be more strict on the version number expected

`metadata upgrade test` branch builder
https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2647/

`StreamsUpgradeTest` branch builder
https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2648/

UPDATE: All branch builders passed

**Review by guozhangwang — APPROVED** [review]
LGTM.

**Comment by bbejeck** [comment]
Merged #6835 into trunk

**Comment by bbejeck** [comment]
cherry-picked #6835 to 2.3

**Pull request merged**